### DISCUSSION TITLE
Official Debian package?

### DISCUSSION THREAD

**Discussion by geckolinux** [Antynea/grub-btrfs#236] [General]
Hi there, I'm very pleased with the functionality that grub-btrfs brings to Linux, and I proudly ship and configure it as the default installation option in SpiralLinux, which is built from Debian Stable. I just wish that it was available as an official Debian package. Has it ever been considered for submission? Now might be a good time before the freeze and eventual release of Debian 12 in 2023. Thanks again for this fantastic project!

**Comment by Schievel1** [comment]
I have no idea how this works process works in Debian. Do you submit it to somewhere or do you contact one of their devs and ask if they are willing to maintain it?

**Reply by geckolinux** [reply]
Hi, thanks for the reply. It looks like an "Intent to package" should be filed:
https://www.debian.org/doc/manuals/distribute-deb/distribute-deb.html#getting-a-sponsor

**Reply by Schievel1** [reply]
Thanks I will have a look.

**Reply by Schievel1** [reply]
So as far as I understand it, we would need to write the packages ourselves and then ask a Debian dev to become a sponsor for this package. Kind of like gentoos proxy maintainers project. 
Well, I do not run Debian or any Debian descendant, so writing and testing packages is kind of a hassle for me. 
You run a Debian based distro. Do you know how to write Debian packages or are you willing to learn it and also willing to write the packages for grub-btrfs?

**Reply by kevinchevreuil** [reply]
Hello, I can help you.
I created a Debian package for the Kaisen Linux distribution find here: https://gitlab.com/kaisenlinux/packages/grub-btrfs

This package is adapted for the integration of other BTRFS tools included in Kaisen Linux. On Kaisen Linux, I developed a tool that uses Timeshift to automatically take snapshots when APT is used.
I developed my own systemd units files in the debian/ directory to automatically add the snapshots taken by timeshift.
I can adapt it for the upstream systemd units files.

What do you think about this?

**Reply by Schievel1** [reply]
are you willing to maintain the Debian package? If so we should file that "Intent to package" and get a Debian dev as sponsor.

Regarding work load this wont be so bad. We usually have releases every few months, last time it was even a year.

**Reply by kevinchevreuil** [reply]
Yes, I can do it.

But, I'm not an official Debian developer. My package may don't respect few Debian packages policies. A Debian developer can reworked it with me to adapt the package to the rules of the Debian project.

So, I'm ready to maintain the official grub-btrfs package, I love this project!

**Reply by Schievel1** [reply]
Awesome! As far as I understand it you get a Debian dev as a sponsor and they will help you get the package to their standards. 
I maintain the package for gentoo. I think you can use this as inspiration when it comes to dependencies. 
https://github.com/gentoo/guru/blob/master/app-backup/grub-btrfs/grub-btrfs-4.12.ebuild

Please note that inotify-tools is listed as optional in the README of us because it is only needed by the grub-btrsd daemon, but the script that does the entries in the grub menu (`/etc/grub.d/41_snapshots-btrfs`) could run without the daemon and therefore without inotify-tools. I put inotify-tools into the dependencies of the ebuild regardless, because most people want to use the daemon and users would just wonder why it does not work. (and they don't read the output of the package manager properly)

Also have a look what I need the patch to meet gentoos QA standard regarding where manpages and docs are installed: https://github.com/gentoo/guru/blob/master/app-backup/grub-btrfs/files/grub-btrfs-9999-remove-docs-from-make.patch
I assume that is similar with Debian, but don't know. 

I will file the intent to package and try to get us a dev to sponsor this.

**Reply by geckolinux** [reply]
Thanks very much to everyone for being open to this idea and to @KaisenCAS for taking it on!
Here's another option that @KaisenCAS can use as a starting point, an unofficial Debian 11 package of `grub-btrfs` on the openSUSE Build Service:
https://build.opensuse.org/package/show/home%3Amkrwc%3Adebian/grub-btrfs

**Reply by Schievel1** [reply]
@KaisenCAS I filed the ITP already, that is: I sent the mail. But I can not find the bug yet. 

I think when you are ready you write the request for sponsorship to their mailing list yourself, attaching your .debian(?) package file. From what I saw they will give suggestions then. 

BTW: I found this while researching if my ITP is already filed: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=941627
Looks like they worked in this already, but somehow this effort fell asleep again. 

Edit: here is the ITP I filed
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1025247

**Reply by Schievel1** [reply]
@geckolinux  @KaisenCAS  please come to the ITP on debian bugzilla. They have some questions regarding testing and such. 
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=941627
